salt/modules/ldap3.py

```diff
@@ -572,19 +572,16 @@ def change(connect_spec, dn, before, after):
     # convert the "iterable of values" to lists in case that's what
     # modifyModlist() expects (also to ensure that the caller's dicts
     # are not modified)
-    before = dict(((attr, list(vals))
+    before = dict(((attr, salt.utils.data.encode(vals))
                    for attr, vals in six.iteritems(before)))
-    after = dict(((attr, list(vals))
+    after = dict(((attr, salt.utils.data.encode(vals))
```

Comment:
I'm not familiar with this module, but `list()` has been there from the beginning (according to git log). Can you explain why you removed it here?

Reply:
In fact, it would be safe to use `salt.utils.data.encode(list(vals))` because vals could be type `OrderedSet`

Reply:
Then you should probably make that change.